src/to_chars.cpp

```diff
@@ -922,8 +923,10 @@ char *to_chars(char *first, const char *last, double value) {
   {
     *first++ = '0';
     // Make it look like a floating-point number (#362, #378)
-    *first++ = '.';
-    *first++ = '0';
+    if(negative) {
```

Comment:
Ah, so the reason here is that we want to write `-0.0` instead of `-0`, so that we trigger double parsers (since -0 can't be represented as an integer). Does this mean we should detect `-0` in our DOM parser and treat it as a double?

Reply:
Done.